src/main/java/org/wildfly/security/auth/server/AbstractMechanismAuthenticationFactory.java

```diff
@@ -39,7 +39,7 @@
         this.mechanismConfigurations = mechanismConfigurations;
     }
 
-    SecurityDomain getSecurityDomain() {
+    public SecurityDomain getSecurityDomain() {
```

Comment:
You'll want to add this to the org.wildfly.security.auth.server.MechanismAuthenticationFactory interface too, otherwise it will not be visible to consumers anyway since the class itself is non-public.